main/calculations.py

```diff
@@ -0,0 +1,147 @@
+    to_email = email
+    send_mail(
+        subject,
+        message,
+        from_email,
+        [to_email],
+        fail_silently=False,
+    )
+
+
+def my_cron_job():
+    users = User.objects.all()
+    for user in users:
+        sites = SiteToCheck.objects.filter(user=user)
+        output = ''
+        for site in sites:
+            if 'bad_data' in SiteDownChecker(site, user).status():
+                output += f'{site} - ERROR\n'
+            elif site.last_status != 200:
+                output += f'{site} - last status: {site.last_status}'
+        if len(output) > 0:
+            send_email(output, user.email)
+
+
+def modify_email(request):
+    response_json = json.dumps(request.POST)
+    data = json.loads(response_json)
+    user = request.user
+    if user.email == data['email']:
+        error_message = 'It is your existing email address'
+        messages.error(request, error_message)
+    else:
+        user.email = data['email']
+        user.save()
+        success_message = f'You changed your email. Every messages will be send to \
+                    {request.user.email} until now.'
+        messages.success(request, success_message)
+
+
+class SiteDownChecker:
+
+    def __init__(self, url, user):
+        self.url = url
+        self.time = 0
+        self.error = None
+        self.user = user
+
+    def status(self, proxy=False):
+        try:
+            if proxy:
+                r = ProxyRequests(self.url)
+            else:
+                r = requests.get(self.url, headers={
+                    'User-Agent': 'Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1)'})
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_success(proxy, r)
+            else:
+                return self.create_new_url_success(proxy, r)
+        except Exception as e:
+            self.error = str(e)
+            if not proxy and config.PROXY:
+                return self.status(proxy=True)
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_exception(e)
+            else:
+                return self.create_url_exception()
+
+    def create_url_exception(self):
+        data = dict()
+        SiteToCheck.objects.create(url=self.url,
+                                   user=self.user,
+                                   last_status=None,
+                                   last_response_time=None,
+                                   last_check=datetime.now().strftime("%Y-%m-%d %H:%M"),
+                                   bad_data=str(
+                                       datetime.now().strftime("%Y-%m-%d %H:%M")) + ': The url is not responding'
+                                   )
+        data['last_status'] = None
+        data['last_response_time'] = None
+        data['bad_data'] = 'The url is not responding'
+        return data
+
+    def modify_url_exception(self, e):
```

Comment:
To by mogło być metodą na modelu SiteToCheck. Tylko trzeba to mądrze nazwać. Zobacz, że operuje ta metoda bardzo mocno na obiekcie modelu `SiteToCheck`. Np. `exception_status`. I ona by zwracał od razu zserializowany błąd.